src/js/common/components/RequiredLoginView.jsx

```diff
@@ -8,111 +8,104 @@ import LoadingIndicator from '../../common/components/LoadingIndicator';
 import Main from '../../login/containers/Main';
 
 const healthTools = ['health-records', 'rx', 'messaging'];
+const nextQuery = { next: window.location.pathname };
+const signInUrl = appendQuery('/', nextQuery);
 
 class RequiredLoginView extends React.Component {
-  isServiceAvailable() {
-    const requiredApp = this.props.serviceRequired;
-    const userServices = this.props.userProfile.services;
-    // Checks that
-    // 1) session has a valid authentication token, and
-    // 2) the application being loaded (requiredApp) is in the list of
-    //    applications/services the user is authorized to use (userServices)
-    // TODO: replace indexOf() once NodeJS versions in all environments support includes()
-
-    if (sessionStorage.userToken && userServices) {
-      if (Array.isArray(requiredApp)) {
-        return intersection(userServices, requiredApp).length > 0;
+  // Checks that (1) session has a valid authentication token and
+  // (2) the user is authorized to use services required by this application
+  isAccessible = () => {
+    const { serviceRequired, user } = this.props;
+    const userServices = user.profile.services;
+    const hasRequiredServices = userServices && (
+      Array.isArray(serviceRequired) ?
+        intersection(userServices, serviceRequired).length > 0 :
+        userServices.includes(serviceRequired)
+    );
+
+    return sessionStorage.userToken && hasRequiredServices;
+  }
+
+  renderVerifiedContent = () => {
+    const { serviceRequired, user } = this.props;
+
+    // Ignore required verification when accessing health tools with MHV sign-in.
+    const hasMhvAccess =
+      healthTools.includes(serviceRequired) &&
+      user.profile.authnContext === 'myhealthevet';
+
+    if (user.profile.verified || hasMhvAccess) {
+      // TODO: Delete the logic around attemptingAppealsAccess once we
+      // resolve the MVI/Appeals Users issues.
+      // if app we are trying to access includes appeals,
+      // bypass the checks for user profile status
```

Comment:
Not sure if this is still relevant and don't know who can confirm.

Reply:
If we no longer have a scenario where appeals should still be accessible even when MVI can't be accessed for the user, I'd like to remove this.

Reply:
Maybe @cvalarida?

Reply:
:thinking: After looking around the code for a bit, I don't know why this was needed for the first iteration of appeals status, so I can't speak confidently to v2. I do know when we make a request to `/v0/appeals`, we send along the user token so when vets-api makes a request to Caseflow, it can send along the appropriate user information. To me, that sounds like we'd need the profile, but this is also how the first iteration worked, so... :man_shrugging: 

@bshyong Git blame says you made this check for the first version of appeals status? Do you have any insight on why it was originally necessary? (Or can you point us to who would have that insight?)

Reply:
@cvalarida I don't remember why this was originally needed; feel free to remove this check if it's irrelevant now